src/main/client/src/components/@commons/FilePicker.tsx

```diff
@@ -0,0 +1,127 @@
+}
+
+class FilePicker extends React.Component<FilePickerProps, FilePickerState> {
+  static defaultProps = {
+    clearButtonLabel: 'Clear',
+    multipleFiles: false,
+    accept: undefined,
+    hideClearButton: false,
+    inputProps: undefined,
+    inputGroupProps: undefined
+  }
+
+  private inputRef = React.createRef<HTMLInputElement>()
+
+  constructor(props: FilePickerProps) {
+    super(props)
+    this.state = {
+      files: null,
+      fileName: ''
+    }
+  }
+
+  componentDidUpdate(_: FilePickerProps, prevState: FilePickerState): void {
+    if (prevState.files !== this.state.files) {
+      const fileArray = new Array<File>()
+      if (this.state.files) {
+        for (const file of this.state.files) {
+          fileArray.push(file)
+        }
+      }
+      this.setState({ ...this.state, fileName: fileArray.map((f) => f.name).join(' & ') })
+      this.props.onFileChange(fileArray)
+    }
+  }
+
+  public reset = (): void => this.handleOnClearClick()
+
+  private handleOnFileChange = (ev: React.ChangeEvent<HTMLInputElement>) => {
+    this.setState({ ...this.state, files: ev.target.files })
+    this.clearInnerInput()
+  }
+
+  private handleOnClearClick = () => {
+    this.setState({ ...this.state, files: null })
+    this.clearInnerInput()
+  }
+
+  private clearInnerInput() {
+    if (this.inputRef?.current) {
+      this.inputRef.current.files = null
+    }
+  }
+
+  private handleOnInputClick = () => {
+    if (this.inputRef?.current) {
+      this.inputRef.current.value = ''
+      this.inputRef.current.click()
+    }
+  }
+
+  render = (): JSX.Element => {
+    const { placeholder, clearButtonLabel, hideClearButton, multipleFiles, accept, inputProps, inputGroupProps } = this.props
+
+    return (
+      <InputGroup {...inputGroupProps}>
+        <input
+          type="file"
+          ref={this.inputRef}
+          accept={accept}
+          style={{ display: 'none' }}
+          multiple={multipleFiles}
+          onChange={this.handleOnFileChange}
+          data-testid={inputProps?.placeholder ?? placeholder}
+        />
+        <Input
+          placeholder={placeholder}
+          {...{
+            ...inputProps,
+            readOnly: true,
+            isReadOnly: true,
+            value: this.state.fileName,
+            onClick: this.handleOnInputClick
+          }}
+        />
+        {!hideClearButton && <ClearButton clearButtonLabel={clearButtonLabel ?? 'Clear'} onButtonClick={this.handleOnClearClick} />}
+      </InputGroup>
+    )
+  }
+}
+
+type ClearButtonProps = Pick<FilePickerProps, 'clearButtonLabel'> & {
+  onButtonClick: () => void
+}
+
+const ClearButton: React.VFC<ClearButtonProps> = ({ clearButtonLabel, onButtonClick }) => (
+  <InputRightAddon>
+    <Button onClick={onButtonClick}>{clearButtonLabel ?? 'Clear'}</Button>
+  </InputRightAddon>
+)
+
+ClearButton.propTypes = {
+  clearButtonLabel: PropTypes.string,
+  onButtonClick: PropTypes.func.isRequired
+}
+
+export default FilePicker
```

Comment:
I know this was copied, but pls rewrite this to `export { FilePicker }` it's better to export every component as objects of the module rather than using default export. (I know, I didn't set this rule in ESLint sorry)

Reply:
But keep it in mind, that this way everywhere where the FilePicker was used, the imports should be rewritten to `import { FilePicker } from './components/@commons/Filepicker'`